### PR TITLE
SoundWire: use inclusive terms

### PR DESCRIPTION
**Pull request by plbossart**
The updated SoundWire specification will be published shortly. The only change is a use of 'inclusive terms'
- Slave->Peripheral
- Master->Manager

Some will note that the change does not preserve the M/S initials - that's known but the MIPI team was unable to find a better set of words.

We do want to remain aligned with the spec so need a massive rename across soundwire/regmap/asoc domains.

I would like to submit this in the 5.14 review cycle, meaning we have about 2 weeks to complete the reviews on GitHub. To avoid making things more complicated than they need to be I just reverted 2 commits related to ASoC codecs that are not upstream. I'll have to rebase this branch if/when they are updated by Realtek/Maxim.

@vinodkoul FYI

### CODE REVIEW

File: Documentation/ABI/testing/sysfs-bus-soundwire-manager [hunk 1]
```diff
@@ -0,0 +1,23 @@
+What:		/sys/bus/soundwire/devices/sdw-manager-N/revision
+		/sys/bus/soundwire/devices/sdw-manager-N/clk_stop_modes
+		/sys/bus/soundwire/devices/sdw-manager-N/clk_freq
+		/sys/bus/soundwire/devices/sdw-manager-N/clk_gears
+		/sys/bus/soundwire/devices/sdw-manager-N/default_col
+		/sys/bus/soundwire/devices/sdw-manager-N/default_frame_rate
+		/sys/bus/soundwire/devices/sdw-manager-N/default_row
+		/sys/bus/soundwire/devices/sdw-manager-N/dynamic_shape
+		/sys/bus/soundwire/devices/sdw-manager-N/err_threshold
+		/sys/bus/soundwire/devices/sdw-manager-N/max_clk_freq
```
Comment (lyakh) [inline Documentation/ABI/testing/sysfs-bus-soundwire-manager]: sdw is relatively new, so, it's possible that nobody is using these properties yet, but we cannot be 100% certain, right? How are such things verified in general? Are (major) distributions and vendors tested or contacted?

**Review by lyakh — COMMENTED** [review]
what's the usual / best process with such scripted changes? Do we just trust that the change is automated and as such must be correct? Isn't it usual to reference the script / command lines that were used for such changes? Reviewers could then at least check the script.

**Review by bardliao — DISMISSED** [review]
The change looks straight forward, so LGTM.

**Comment by plbossart** [comment]
> what's the usual / best process with such scripted changes? Do we just trust that the change is automated and as such must be correct? Isn't it usual to reference the script / command lines that were used for such changes? Reviewers could then at least check the script.

it's not fully scripted.
I used a standard s/slave/peripheral/g and s/master/manager/g but also checked the results. E.g. there are existing firmware properties such as 'master-count' that we cannot rename since that would break the solution. Likewise some 's_rt' variables were renamed as 'peri_rt' to have a better mapping between variable names and types. There are also indentation changes needed when the name of variable changes.

**Review by RanderWang — DISMISSED** [review]
LGTM

**Review by kv2019i — DISMISSED** [review]
The sysfs renames are a bit in the grey area, but given there is no documented usage anywhere, this will probably go through. Otherwise looks good to me, you even fixed all the diagram indents, wow! :D

**Review by RanderWang — DISMISSED** [review]
LGTM

**Comment by bardliao** [comment]
@plbossart Do we need to change the terms of device property, too? For example, "mipi-sdw-master-count", "mipi-sdw-slave-channelprepare-timeout"?

**Review by bardliao — DISMISSED** [review]

**Comment by plbossart** [comment]
> @plbossart Do we need to change the terms of device property, too? For example, "mipi-sdw-master-count", "mipi-sdw-slave-channelprepare-timeout"?

No, this has to stay for backwards-compatibility. There is a suggested rename in MIPI circles but we have to keep supporting older specs.

**Review by RanderWang — DISMISSED** [review]

**Review by RanderWang — DISMISSED** [review]

**Comment by aiChaoSONG** [comment]
I think term revision PR should be merged asap, or the author will need to rebase again and again..., which is time consuming.

**Comment by plbossart** [comment]
> I think term revision PR should be merged asap, or the author will need to rebase again and again..., which is time consuming.

I know, but we always have a good reason to push this out. This time it's the revision of registers and the split between shim and device in PR #3811

**Comment by RanderWang** [comment]
how about sound/soc/intel/boards/*sdw* ? I can't find it in PR since many changes

**Comment by plbossart** [comment]
The last update was a year ago. It pains me to write this but I don't have the time to update this PR and with all the recent SoundWire changes it's become really difficult to make this inclusive terminology change at the 'right' time.

I'll close this for now, we can only restart later but it's a shame we couldn't get this done earlier. FYI @bardliao @lgirdwood

**Pull request closed without merging**